pkg/oc/cli/cmd/startbuild.go

```diff
@@ -312,6 +304,33 @@ func (o *StartBuildOptions) Complete(f *clientcmd.Factory, in io.Reader, out, er
 	return nil
 }
 
+// Validate returns validation errors regarding start-build
+func (o *StartBuildOptions) Validate() error {
+	if o.AsBinary {
+		if len(o.LogLevel) > 0 {
+			fmt.Fprintf(o.ErrOut, "WARNING: Specifying --build-loglevel with binary builds is not supported.\n")
+		}
+		if len(o.EnvVar) > 1 || (len(o.LogLevel) == 0 && len(o.EnvVar) > 0) {
+			fmt.Fprintf(o.ErrOut, "WARNING: Specifying environment variables with binary builds is not supported.\n")
+		}
+		if len(o.BuildArgs) > 0 {
+			fmt.Fprintf(o.ErrOut, "WARNING: Specifying build arguments with binary builds is not supported.\n")
+		}
+	}
+
+	if len(o.FromBuild) != 0 && o.AsBinary {
+		// TODO: we should support this, it should be possible to clone a build to run again with new uploaded artifacts.
+		// Doing so requires introducing a new clonebinary endpoint.
+		return fmt.Errorf("Cannot use '--from-build' flag with binary builds")
+	}
+
+	if len(o.Name) == 0 && len(o.FromWebhook) == 0 && len(o.FromBuild) == 0 {
```

Comment:
This is still not accurate actually. `len(o.Name) > 0 && (len(o.FromBuild) > 0` may also need to produce warn or error. But, it will be a new validate, so I leave this as these empty validations only.